rpcmetrics/span.go

```diff
@@ -0,0 +1,199 @@
+package rpcmetrics
+
+import (
+	"sync"
+	"time"
+
+	"strconv"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/opentracing/opentracing-go/log"
+)
+
+// SpanKind identifies the span as inboud, outbound, or internal
+type SpanKind int
+
+const (
+	// Local span kind
+	Local SpanKind = iota
+	// Inbound span kind
+	Inbound
+	// Outbound span kind
+	Outbound
+)
+
+// Span is a wrapper around opentracing.Span that collects RPC metrics
+type Span struct {
+	tracer         *Tracer
+	realSpan       opentracing.Span
```

Comment:
I find those extremely confusing. The decorator span is also wrapped around the real span, so which one is "wrapped" span?